src/main/java/Parser.java

```diff
@@ -0,0 +1,174 @@
+            String processedTime = "";
+            if (dates.length == 1) {
+                LocalDate localDate = LocalDate.parse(dates[0]);
+                processedDate = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+                return processedDate;
+            } else {
+                LocalDate localDate = LocalDate.parse(dates[0]);
+                processedDate = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+                double time = Double.parseDouble(dates[1]);
+                if (time >= 1300) {
+                    time -= 1200;
+                }
+                int hour = (int) (time / 100);
+                if (hour == 24 || hour == 0) {
+                    processedTime += "12.";
+                } else {
+                    processedTime += hour + ".";
+                }
+                int minute = (int) (time % 100);
+                if (minute < 10) {
+                    processedTime += "0" + minute;
+                } else {
+                    processedTime += minute;
+                }
+                if ((Double.parseDouble(dates[1]) >= 1200)
+                        && (Double.parseDouble(dates[1]) < 2400)) {
+                    processedTime += "pm";
+                } else {
+                    processedTime += "am";
+                }
+                return processedDate + " " + processedTime;
+            }
+        } catch (java.time.format.DateTimeParseException e) {
+            throw new DateException("Sorry! I don't understand the date/time. Please specify the date/time "
+                    + "in YYYY-MM-DD or YYYY-MM-DD HHMM format.");
+        }
+    }
+
+    /**
+     * Parses the user command.
+     * @param s User command.
+     * @param size Current size of the task list.
+     * @return List of String objects that represent the parsed user command.
+     * @throws InvalidDoneException If for a done command, either no task is
+     * specified or the argument provided does not represent a valid task.
+     * @throws InvalidTaskArgumentException If the command to add tasks is not
+     * correctly or completely specified.
+     * @throws InvalidDeleteException If for a delete command, either no task is
+     * specified or the argument provided does not represent a valid task.
+     * @throws InvalidCommandException If the user command cannot be understood.
+     * @throws DateException If the given date/time for an event/deadline is not
+     * in a format appropriate for parsing.
+     */
+    public ArrayList<String> parseString(String s, int size)
+            throws InvalidDoneException, InvalidTaskArgumentException,
+            InvalidDeleteException, InvalidCommandException, DateException {
+        ArrayList<String> lst = new ArrayList<>();
+        if (s.equals("list")) {
+            lst.add("Show");
+            return lst;
+        } else if ((s.length() >= 4) && (s.substring(0, 4).equals("done"))) {
+            if (s.length() <= 5) {
```

Comment:
This logic seems to depend on the done command only having 4 letters. What if the done command changes to another word?

Reply:
Hmm, for this, I am assuming that the user types in "done" followed by the number of the task to be marked as done, that's why I identify the command by the first four words